.github/workflows/docker-build.yml

```diff
@@ -0,0 +1,46 @@
+name: Build docker image
+
+on:
+  push:
+    tags: "*"
+    branches: ["main", "ci-test"]
+
+permissions:
+  packages: write
+
+jobs:
+  image:
+    name: Build docker image
+    runs-on: [ubuntu-20.04]
+    concurrency:
+      group: ${{ github.ref }}
+      cancel-in-progress: true
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Docker login for ghcr
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{github.actor}}
+          password: ${{github.token}}
+
+      - name: Set image tag for release
+        id: tag
+        run: |
+          if [[ ${{github.ref_type}} == tag ]]; then
+            echo "tag=${{github.ref_name}}"
+          else
+            echo "tag=dev"
+          fi >>$GITHUB_OUTPUT
+
+      - name: Build docker image
+        env:
+          REGISTRY: ghcr.io/kubewharf
+          TAG: ${{ steps.tag.outputs.tag }}
+          ARCHS: amd64
+          OUTPUT_TYPE: registry
```

Comment:
Use the `registry` OUTPUT_TYPE, which is shorthand for `type=image,push=true`. But there is also github action that perform push below, and it seems that two push operations have been performed.

Reply:
According to my previous test, `registry` only tags the image without pushing.

Reply:
Oh it might be related to the comment below.